sdk/identity/azure-identity/inc/azure/identity/detail/token_cache.hpp

```diff
@@ -24,21 +25,17 @@ namespace Azure { namespace Identity { namespace _detail {
    * @brief Access token cache.
    *
    */
-  class TokenCache
-#if !defined(TESTING_BUILD)
-      final
-#endif
-  {
-#if !defined(TESTING_BUILD)
+  class TokenCache _azure_NON_FINAL_FOR_TESTS {
+#if !defined(_azure_TESTING_BUILD)
   private:
 #else
   protected:
 #endif
```

Comment:
This, I think, can be avoided, and be made always private.

Reply:
I wanted to do that, but there are some tests that want access to the `CacheValue` struct/etc., so kept it as protected for now. We can clean that up and make more things "friends" later, but given this is in `detail` as implementation detail only, I deprioritized it.
https://github.com/Azure/azure-sdk-for-cpp/blob/61f5ce00c2908d9017b2ad3e7891ced99fe67de0/sdk/identity/azure-identity/test/ut/token_cache_test.cpp#L18-L20

Reply:
Done